src/api/java/com/minecolonies/api/items/IBlockOverlayItem.java

```diff
@@ -0,0 +1,31 @@
+package com.minecolonies.api.items;
+
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.item.ItemStack;
+import net.minecraft.world.level.Level;
+import net.minecraft.world.phys.AABB;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.List;
+
+/**
+ * An interface to be implemented by items that want to render overlays while the player is holding the item.
+ */
+public interface IBlockOverlayItem
+{
+    /**
+     * Called client-side only.
+     * @return a list of overlay boxes that should be rendered for this item.
+     */
+    @NotNull
+    List<OverlayBox> getOverlayBoxes(@NotNull final Level world, @NotNull final Player player, @NotNull final ItemStack stack);
+
+    /**
+     * Details about the overlay box to draw.
+     * @param bounds  the bounds of the box.
+     * @param color   the line color.
+     * @param width   the line width.
+     * @param clipped false to display through blocks.

```

Comment:
I would go for sth more descriptive, clipped is weird param name, vanilla has seeThrough or sth similar would be better. Or enum?

Reply:
The box isn't seeThrough, the block is.  So that's the wrong perspective.

Reply:
well look at font rendering mode xd
still clipped is for scissoring etc. not for no_depth_test thingiest

Reply:
I've renamed it to `showThroughBlocks`.